model/src/test/kotlin/VcsInfoTest.kt

```diff
@@ -145,6 +145,31 @@ class VcsInfoTest : StringSpec({
             inputA.merge(inputB) shouldBe output
         }
 
+        "prefer more complete information for GitLab" {
+            val inputA = VcsInfo(
+                    provider = "",
+                    url = "https://gitlab.com/rich-harris/rollup-plugin-buble.git",
+                    revision = "",
+                    path = ""
+            )
+
+            val inputB = VcsInfo(
+                    provider = "git",
+                    url = "git+https://gitlab.com/rich-harris/rollup-plugin-buble.git",
+                    revision = "9928a569351a80c2f7dc065f61085954daed5312",
+                    path = ""
+            )
+
+            val output = VcsInfo(
+                    provider = "git",
+                    url = "https://gitlab.com/rich-harris/rollup-plugin-buble.git",
+                    revision = "9928a569351a80c2f7dc065f61085954daed5312",
+                    path = ""
+            )
+
+            inputA.merge(inputB) shouldBe output
+        }
+
```

Comment:
It would also be helpful to have a test that verifies what happens when VcsInfos with different providers are merged (e.g. never take the URL/revision from one and the provider from the other).

Reply:
Agreed, but let's leave that as a future improvement to get this code it sooner 😉